src/basic/memory-util.c

```diff
@@ -39,3 +39,19 @@ bool memeqbyte(uint8_t byte, const void *data, size_t length) {
         /* Now we know first 16 bytes match, memcmp() with self.  */
         return memcmp(data, p + 16, length) == 0;
 }
+
+void *make_swapped(const void *data, size_t size) {
+        assert(data);
+        assert(size != 0);
+
+        void *p = malloc(size);
+        if (!p)
+                return NULL;
+
+        uint8_t *p_dst = p;
+        const uint8_t *p_src = data;
+        for (size_t i = 0, k = size - 1; i < size; i++, k--)
```

Comment:
doesn't matter much, but I think it sucks a bit that k is definitely going to underflow here. I mean, I am aware it's not going to be used if it underflows, but it still underflows here.

maybe avoid this, by initializing k to `size` rather than `size-1` here, and then changing the loop to do `p_dst[i] = p_src[j-1];`, i.e. subtract one there.

That way we never underflow...

Reply:
Don't see a problem here, but OK. I think a compiler will shift the index anyway.